### DISCUSSION TITLE
Why does Budibase need "Third-Party Cookies" to be allowed?

### DISCUSSION THREAD

**Discussion by pandel** [Budibase/budibase#7324] [Help] [answered]
Hi!

We use Firefox in our environment and third-party cookies are being blocked via policy by default. That leads to a problem with Budibase, as it is only working correctly if I unblock them. If I don't I get many error messages and a non-functioning app designer.

For testing purpose this is ok, but I cannot activate third-party cookies in general for every user, so why are these cookies necessary for Budibase and how can I avoid them? I have no other internal web application that is depending on third-party cookies, so I have no experience with this situation...

I get a lot of "Cookie denied" messages in the browser console. Here are some examples. As my messages are in German I can only try to translate them:

`Cookie “budibase:auth.sig” will soon be handled as Cross-Site-Cookie against “https://<servername>:10000/api/queries" because the schema does not match`
or 
`Third-Party Cookie "" denied. 2 [remixicon.css](https://cdn.jsdelivr.net/npm/remixicon@2.5.0/fonts/remixicon.css)`
or
`Third-Party Cookie "budibase:currentapp=; path=/; expires=Tue, 19 Jan 2038 03:14:07 GMT" denied.`

Regards,
Holger

**Comment by aptkingston** [comment, marked as answer]
Hey @pandel. Third party cookies are only required for developers who will be using the builder, and actually creating applications. They aren't required for normal end-users who will just use the built applications.

The reason they're required for developer users is because of how the app preview works when you're inside the builder. The app preview is an iframe with a `srcdoc` attribute. Becuase of this it effectively has no valid origin, therefore considering *any* cookies as third party. There's no workround for this at the moment - they're just required in order to be able to use the app preview.

Third party cookies should not be required at all for the majority of users though.

I have some ideas around how to fix this requirement, so hopefully this will change soon.

**Reply by aptkingston** [reply]
PR here: https://github.com/Budibase/budibase/pull/7383

**Reply by pandel** [reply]
@aptkingston
Many thanks for the detailed explanation, I understand. And Wow! for the PR. Hopefully it can be merged soon...
